File: 0985-sum-of-even-numbers-after-queries/0985-sum-of-even-numbers-after-queries.py
class Solution:
    def sumEvenAfterQueries(self, nums: List[int], queries: List[List[int]]) -> List[int]:
        res = []
        even_nums = [x for x in nums if x%2 == 0]

        for q in queries:
            if nums[q[1]] % 2 == 0:
                even_nums.remove(nums[q[1]])
                
            nums[q[1]] += q[0]

            if nums[q[1]] % 2 == 0:
                even_nums.append(nums[q[1]])
            
            res.append(sum(even_nums))
        
        return res 

File: 0985-sum-of-even-numbers-after-queries/0985-sum-of-even-numbers-after-queries.go
func sumEvenAfterQueries(nums []int, queries [][]int) []int {
		res := make([]int, len(queries))
		sum := 0
		for _, num := range nums {
			if num%2 == 0 {
				sum += num
			}
		}
		for i, q := range queries {
			val, idx := q[0], q[1]
			if nums[idx]%2 == 0 {
				sum -= nums[idx]
			}
			nums[idx] += val
			if nums[idx]%2 == 0 {
				sum += nums[idx]
			}
			res[i] = sum
		}
		return res
}